benchmarks/utils/benchmark_ts.sh

```diff
@@ -0,0 +1,164 @@
+    fi
+
+    python ts_scripts/install_from_src.py
+
+    pip install -r benchmarks/requirements-ab.txt
+fi
+
+# generate benchmark json config files
+rm -rf json
+mkdir -p json/cpu
+mkdir -p json/gpu
+
+for model in "${models[@]}"; do
+    input_file="./benchmarks/models_config/${model}"
+    echo "input_file=$input_file"
+    python ./benchmarks/utils/gen_model_config_json.py --input $input_file --output json
+done
+
+# run benchmark
+config_dir="json/cpu"
+if [ "$hw_type" == "GPU" ]; then
+    config_dir="json/gpu"
+fi
+
+for config_file in "$config_dir"/*; do
+    echo "config_file=$config_file"
+    if [ -f "$config_file" ]; then
+        rm -rf ./logs
+        python ./benchmarks/benchmark-ab.py --config_properties ./benchmarks/config.properties --config $config_file
+
+	      model_name=`echo $config_file |cut -d'/' -f 3|cut -d'.' -f 1`
+
+	      python ./benchmarks/utils/gen_metrics_json.py --csv /tmp/benchmark/ab_report.csv \
+	      --log /tmp/benchmark/logs/model_metrics.log \
+	      --stats /tmp/benchmark/logs/stats_metrics.json \
+	      --raw /tmp/benchmark/logs/model_metrics.json
+
+	      if [ "$3" == "nightly" ]; then
+            aws cloudwatch put-metric-data \
+            --namespace torchserve_benchmark_${hw_type} \
+            --region "us-west-2" \
+            --metric-data \
+            file:///tmp/benchmark/logs/stats_metrics.json
+        fi
+
+	      mkdir -p /tmp/ts_benchmark/${model_name}
+	      if [ -f /tmp/benchmark/ab_report.csv ]; then
+            mv /tmp/benchmark/ab_report.csv /tmp/ts_benchmark/${model_name}/ab_report.csv
+        fi
+	      if [ -f /tmp/benchmark/predict_latency.png ]; then
+            mv /tmp/benchmark/predict_latency.png /tmp/ts_benchmark/${model_name}/predict_latency.png
+        fi
+	      if [ -f /tmp/benchmark/result.txt ]; then
+            mv /tmp/benchmark/result.txt /tmp/ts_benchmark/${model_name}/result.txt
+        fi
+	      if [ -f /tmp/benchmark/logs/model_metrics.log ]; then
+            mv /tmp/benchmark/logs/model_metrics.log /tmp/ts_benchmark/${model_name}/model_metrics.log
+        fi
+        if [ -f /tmp/benchmark/logs/model_metrics.json ]; then
+            mv /tmp/benchmark/logs/stats_metrics.json /tmp/ts_benchmark/${model_name}/stats_metrics.json
+        fi
+        if [ -f ./logs/model_log.log ]; then
+            mv ./logs/model_log.log /tmp/ts_benchmark/${model_name}/model_log.log
+        fi
+        if [ -f ./logs/ts_log.log ]; then
+            mv ./logs/ts_log.log /tmp/ts_benchmark/${model_name}/ts_log.log
+        fi
+
+    fi
+done
+
+# generate final report
+python ./benchmarks/utils/gen_md_report.py \
+  --input /tmp/ts_benchmark/ \
+  --output /tmp/ts_benchmark/report.md \
+  --hw ${hw_type} \
+  --branch ${BRANCH}
+
+# clean up
+rm -rf json
+git checkout master
+
+# save to S3
+if [[ "$3" == "nightly" ]]; then
+    dt=$(date +'%Y-%m-%d')
+    aws s3 cp --recursive /tmp/ts_benchmark/ s3://torchserve-model-serving/benchmark/${dt}/${hw_type,,}/
```

Comment:
I'd much rather we save this in github artifacts so the experience is fully within github, we can integrate this as a github action

Also community members shouldn't be able to upload to an S3 bucket primarily managed by torchserve team

Reply:
let's add a comment here as well then to make this clearer